src/components/Chat/ShareModal.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  FloatingFocusManager,
+  FloatingOverlay,
+  FloatingPortal,
+  useDismiss,
+  useFloating,
+  useInteractions,
+} from '@floating-ui/react';
+import { IconCopy, IconX } from '@tabler/icons-react';
+import { IconCheck } from '@tabler/icons-react';
+import { MouseEvent, useCallback, useRef, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import { Tooltip, TooltipContent, TooltipTrigger } from '../Common/Tooltip';
+
+import { v4 as uuidv4 } from 'uuid';
+
+export enum SharingType {
+  Conversation = 'conversation',
+  ConversationFolder = 'conversations_folder',
+  Prompt = 'prompt',
+  PromptFolder = 'prompts_folder',
+}
+
+interface Entity {
+  id: string;
+  name: string;
+}
+
+interface Props {
+  entity: Entity;
+  type: SharingType;
+  isOpen: boolean;
+  onClose: () => void;
+  onShare: (shareId: string) => void;
+}
+
+export default function ShareModal({
+  entity,
+  isOpen,
+  onClose,
+  onShare,
+  type,
+}: Props) {
+  const { t } = useTranslation('sidebar');
+  const copyButtonRef = useRef<HTMLButtonElement>(null);
+  const [urlCopied, setUrlCopied] = useState(false);
+  const [urlWasCopied, setUrlWasCopied] = useState(false);
+  const shareId = uuidv4();
+  const url = `${window?.location.origin}/share/${shareId}`;
+
+  const { refs, context } = useFloating({
+    open: isOpen,
+    onOpenChange: () => {
+      onClose();
+    },
+  });
+  const dismiss = useDismiss(context);
+  const { getFloatingProps } = useInteractions([dismiss]);
+
+  const handleClose = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      onClose();
+    },
+    [onClose],
+  );
+
+  const handleCopy = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+      if (!navigator.clipboard) return;
+
+      navigator.clipboard.writeText(url).then(() => {
+        setUrlCopied(true);
+        setTimeout(() => {
+          setUrlCopied(false);
+        }, 2000);
+        if (!urlWasCopied) {
+          setUrlWasCopied(true);
+          onShare(shareId);
+        }
+      });
+    },
+    [onShare, shareId, url, urlWasCopied],
+  );
+
+  return (
+    <FloatingPortal id="theme-main">
+      <FloatingOverlay
+        lockScroll
+        className="z-50 flex items-center justify-center bg-gray-900/30 p-3 dark:bg-gray-900/70 md:p-5"
+      >
+        <FloatingFocusManager context={context} initialFocus={copyButtonRef}>
+          <form
+            noValidate
+            className="relative inline-block max-h-full w-full max-w-[424px] gap-4 rounded bg-gray-100 p-6 text-left dark:bg-gray-700"
```

Comment:
nit: Is 424 px height is intended? Maybe it's better to round to something like 425? :)

Reply:
Darya said 424px :)